pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -632,7 +630,16 @@ func deleteTransitionMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterD
 }
 
 // logCumulativeMetric should be used to log cumulative metrics
-func logCumulativeMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment, time float64) {
+// Time to log is calculated as the time lapsed since the last transition time of condition mentioned
+func logCumulativeMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment,
+	conditionType hivev1.ClusterDeploymentConditionType, logger log.FieldLogger) {
+	condition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, conditionType)
+	// This shouldn't happen if conditions have been properly initialized
+	if condition == nil{
+		logger.Warningf("cannot find %s condition, logging of %s metric skipped", conditionType, &metric)
```

Comment:
A few things here:
- This represents a code bug for sure, so I would a) definitely make it an Error rather than a Warning; b) probably even put a loud eyecatcher in the message, e.g. `THIS IS A BUG!`.
- Isn't `&metric` a pointer to a pointer? So you'll end up with an unhelpful hex address in the message? Did you mean `*metric`?
- Prefer structured logging over embedded formats: `logger.WithField("conditionType", conditionType).WithField("metric", *metric).Error("cannot find condition, logging of metric skipped")`